chartfx-chart/src/test/java/io/fair_acc/chartfx/utils/NumberFormatterImplTest.java

```diff
@@ -0,0 +1,120 @@
+package io.fair_acc.chartfx.utils;
+
+import org.junit.jupiter.api.Test;
+
+import java.text.DecimalFormatSymbols;
+import java.util.Locale;
+import java.util.function.DoubleFunction;
+
+import static io.fair_acc.chartfx.utils.NumberFormatterImpl.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test cases based on removed com.ibm.icu.text.DecimalFormat
+ *
+ * @author Florian Enner
+ * @since 07 March 2023
+ */
+class NumberFormatterImplTest {
+
+    @Test
+    void plainFormatGerman() {
+        Locale.setDefault(Locale.GERMAN);
+        var formatter = createFormatter(false, ALL_DIGITS);
+        assertEquals("0", formatter.apply(0));
+        assertEquals("2", formatter.apply(2.0));
+        assertEquals("2,1", formatter.apply(2.1));
+        assertEquals("10", formatter.apply(10));
+        assertEquals("123,456789", formatter.apply(123.456789));
+        assertEquals("1,23456789E14", formatter.apply(123.456789E12));
+        assertEquals("0,000000000123456789", formatter.apply(123.456789E-12));
+    }
+
+    @Test
+    void plainFormatEnglish() {
+        Locale.setDefault(Locale.US);
+        var formatter = createFormatter(false, ALL_DIGITS);
+        assertEquals("0", formatter.apply(0));
+        assertEquals("2", formatter.apply(2.0));
+        assertEquals("2.1", formatter.apply(2.1));
+        assertEquals("10", formatter.apply(10));
+        assertEquals("123.456789", formatter.apply(123.456789));
+        assertEquals("1.23456789E14", formatter.apply(123.456789E12));
+        assertEquals("0.000000000123456789", formatter.apply(123.456789E-12));
```

Comment:
It bugs me a bit that the behavior is so different for positive and negative exponents, intuitively I would expect them to behave in a symmetric way. Is there an easy rationale to why it has to be that way?
I realize that these are pathological cases that will never be triggered in the label code due to the way the labels are generated, but some plugins or users might want to do number formatting consistent with the labels and feed the formatter arbitrary data (e.g. the value indicators).

Reply:
The byte buffer has a limited size that would not be able to fit very large numbers (e.g. Double.MAX_VALUE would need 300 zero digits). It took me too long to figure out the special casing, and I didn't want to waste buffer space for something that'll never happen, so I added an arbitrary threshold at which it just uses the exponential form.

I originally had a similar threshold for negative exponents, but I added the required special casing to support zooming around zero without changing formats.